### PR TITLE
rpcdaemon: fix on create failure trace

### PR DESCRIPTION
**Pull request by Sixtysixter**
Fix on create failure trace
To be merged after [rpc-test PR](https://github.com/erigontech/rpc-tests/pull/163) merge and tag

### CODE REVIEW

File: silkworm/core/execution/evm.cpp [hunk 1]
```diff
@@ -97,10 +97,16 @@ CallResult EVM::execute(const Transaction& txn, uint64_t gas) noexcept {
 
 evmc::Result EVM::create(const evmc_message& message) noexcept {
     evmc::Result res{EVMC_SUCCESS, message.gas, 0};
+    const evmc_revision rev{revision()};
```
Comment (canepat) [inline silkworm/core/execution/evm.cpp:100]: Moving this declaration here is not necessary and decreases readability, so I would leave it where it was (see below)

File: silkworm/rpc/core/evm_trace.cpp [hunk 2]
```diff
@@ -888,9 +892,38 @@ void TraceTracer::on_execution_end(const evmc_result& result, const silkworm::In
                << " gas_left: " << std::dec << result.gas_left;
 }
 
+void TraceTracer::on_creation_check_failed(const evmc_result& result, const evmc_message& msg) noexcept {
```
Comment (canepat) [inline silkworm/rpc/core/evm_trace.cpp:895]: After we change the callback name to `on_pre_check_failed`, here you can determine anyway the correct value for `trace.type` just using `msg.kind`.

**Review by canepat — CHANGES_REQUESTED** [review]

File: silkworm/core/execution/evm.cpp (hunk 1 above)
Comment (Sixtysixter) [reply silkworm/core/execution/evm.cpp:100]: Done

File: silkworm/rpc/core/evm_trace.cpp (hunk 2 above)
Comment (Sixtysixter) [reply silkworm/rpc/core/evm_trace.cpp:895]: Done

File: silkworm/core/execution/evm.cpp [hunk 3]
```diff
@@ -97,10 +97,16 @@ CallResult EVM::execute(const Transaction& txn, uint64_t gas) noexcept {
...
 
     auto value{intx::be::load<intx::uint256>(message.value)};
     if (state_.get_balance(message.sender) < value) {
         res.status_code = EVMC_INSUFFICIENT_BALANCE;
+
+        for (auto tracer : tracers_) {
+            tracer.get().on_creation_check_failed(res.raw(), message);
```
[the remark this answers is not in the document]
Comment (Sixtysixter) [reply silkworm/core/execution/evm.cpp:107]: Done

**Review by canepat — APPROVED** [review]

**Pull request merged**